File: week_02/jwlee/Digital_root.java
public class DRoot {
  public static int digital_root(int n) {
    
      
      int sum = 0;
      
    while(n >= 10) {
       sum += n % 10;  
       n /= 10;
    }
      sum += n;
      
    return sum >= 10 ? digital_root(sum) : sum;
   
}
  }

File: week_01/code_wars/cmSon/Digital_Root.cpp
#include <iostream>

using namespace std;

int digital_root(int n)
{
    int check_digit = n  / 10;
  
    if( check_digit == 0){
      return n;
    }else{
      int sum = 0;
      
      while (n != 0){
        sum += n % 10;
        n /= 10;
      }
           
      return digital_root(sum);
    }
  
}